src/utils/parse-utils.js

```diff
@@ -14,17 +15,19 @@ export default async function ProcessSpec(specUrl, resolveCircularRefs){
 
   let refParserOptions = {
     resolve: {
-      http: { 
-        withCredentials: false 
-      } 
+      http: {
+        withCredentials: false
+      }
     }
   };
   try {
     if (typeof specUrl==="string") {
-      convertedSpec = await converter.convertUrl(specUrl, convertOptions);
+      const resolvedSpec = await JsonRefs.resolveRefsAt(specUrl, {resolveCirculars: resolveCircularRefs==='true'});
+      convertedSpec = await converter.convertObj(resolvedSpec.resolved, convertOptions);
     }
     else {
-      convertedSpec = await converter.convertObj(specUrl, convertOptions);
+      const resolvedSpec = await JsonRefs.resolveRefs(specUrl, {resolveCirculars: resolveCircularRefs==='true'});
+      convertedSpec = await converter.convertObj(resolvedSpec.resolved, convertOptions);
     }
 
     if (resolveCircularRefs==='false'){
```

Comment:
You dont need this` if` and the `else` block at all, if you are dereferencing using `json-refs`.  `parser` in this code  is an instance of `json-schema-ref-parser` and `parser.dereference` is same as `JsonRefs.resolveRefs`

Now with `json-schema-ref-parser`, for certain It fails to dereference properly and leads to  non-responsive browser, therefore as a workaround there is an attribute provided at the `<rapi-doc>` element `resolve-circular-refs`
its default value is `'true'` but in case some spec is hanging up, then user of the `<rapi-doc>` element can decide not to resolve the refs, in that case we use json-schema-ref-parser's `bundle()` method which wont resolve the refs. This will at least show up the spec in the UI but wherever it has `$ref` it will try to lazily resolve

So remove this block for now, `resolve-circular-refs` attribute too will not have any purpose, if you want to clean up that code then you need to start from `rapidoc.js` and then it is passed down to the lower components, if you want to skip it, thats ok, I will do it later.

Now `parser` object being redundant should be removed, but it is passed to lower components for lazy resolving of `$refs`, refer to line `188` of this file where it is passed this too needs to be cleaned up, else we wont be able to get rid of  `json-schema-ref-parser`.

Goal should be to use either of the parser not both and keep our dependencies and bundle size to minimum and.

Reply:
I am accepting the PR for now, and let me slowly remove things that are not needed.
Thanks for your contribution, 

`json-refs` is indeed better than `json-schema-ref-parser` when it comes to handling large specs. but it increases the bundle size from `98 kb` to `140 kb` (gzipped) but it is a better tradeoff